object_gomap_test.go

```diff
@@ -430,17 +435,40 @@ func TestGoMapMemUsage(t *testing.T) {
 					"test": &nestedMap,
 				},
 			},
+			memLimit: 100,
 			// overhead + len("testN") with string overhead + nested overhead
 			expectedMem: SizeEmptyStruct + (4 + SizeString) + SizeEmptyStruct,
 			// overhead + len("testN") with string overhead + nested overhead
 			expectedNewMem: SizeEmptyStruct + (4 + SizeString) + SizeEmptyStruct,
 			errExpected:    nil,
 		},
+		{
+			name: "should exit early when exceeding memory limit",
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test0": valueInt(99),
+					"test1": valueInt(99),
+					"test2": valueInt(99),
+					"test3": valueInt(99),
+					"test4": valueInt(99),
+					"test5": valueInt(99),
+				},
+			},
+			memLimit: 0,
```

Comment:
[q] Does zero work for this edge case? We don't just exit early. I would think there are cases where we check for memlimit zero.

Reply:
It works as expected here, pretty much as soon as we enter the loop we are exceeding the limit, I can increase this to include 2 values but it doesn't change much. You can see the expected memory accounts for at least one entry to be accounted for.

Reply:
🖖 Sounds good just wanted to make sure there weren't any potential `== 0` checks that might make this test void.